### PR TITLE
Add additional error handling to email message sending.

### PR DESCRIPTION
**Pull request by simongui**
This PR adds some additional error handling where we were ignoring errors when sending emails from the Satellite.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: internal/post/message.go [hunk 1]
```diff
@@ -70,14 +70,24 @@ func (msg *Message) Bytes() []byte {
 		var sub io.Writer
 
 		if len(msg.PlainText) > 0 {
-			sub, _ = wr.CreatePart(textproto.MIMEHeader{
+			sub, err := wr.CreatePart(textproto.MIMEHeader{
 				"Content-Type":              []string{"text/plain; charset=UTF-8; format=flowed"},
 				"Content-Transfer-Encoding": []string{"quoted-printable"},
 			})
+			if err != nil {
+				return nil, err
```
Comment (egonelbre) [inline internal/post/message.go]: Let's also use error wrapping to be able to trace back to the location where the error happened.

File: internal/post/message.go [hunk 2]
```diff
@@ -70,14 +70,24 @@ func (msg *Message) Bytes() []byte {
...
 
 			enc := quotedprintable.NewWriter(sub)
-			_, _ = enc.Write([]byte(msg.PlainText))
-			_ = enc.Close()
+			_, err = enc.Write([]byte(msg.PlainText))
+			if err != nil {
+				return nil, err
+			}
+
+			err = enc.Close()
```
[the remark this answers is not in the document]
Comment (simongui) [reply internal/post/message.go]: That would hide the error. That `enc.Close()` function can return an error. I don't think we want to silently ignore those.

[the remark this answers is not in the document]
Comment (thepaul) [reply internal/post/message.go]: you can still capture an error inside defer. one common pattern in our code is

```go
    defer func() { err = errs.Combine(err, thing.Close()) }()
```

[the remark this answers is not in the document]
Comment (simongui) [reply internal/post/message.go]: Ah interesting! I never thought of doing it that way. What happens if in later code before the defer executes an error happens? Is that what the `errs.Combine()` call is for to combine the first error with the error happening in the defer call?

[the remark this answers is not in the document]
Comment (ifraixedes) [reply internal/post/message.go]: > Combine combines multiple errors into a single error.

It's the doc text of such function https://godoc.org/github.com/zeebo/errs#Combine

It's missing that if `err` are nil they are skipped.

File: internal/post/message.go [hunk 3]
```diff
@@ -93,18 +105,26 @@ func (msg *Message) Bytes() []byte {
 			fmt.Fprint(sub, part.Content)
 		}
 
-		_ = wr.Close()
+		err := wr.Close()
```
Comment (ifraixedes) [inline internal/post/message.go]: This should be deferred after the assignation so guarantee that it always gets close.

Use the same pattern than above `defer func() { err = errs.Combine(err, ...) }()`

File: internal/post/message.go [hunk 4]
```diff
@@ -70,14 +75,21 @@ func (msg *Message) Bytes() []byte {
...
 
 			enc := quotedprintable.NewWriter(sub)
-			_, _ = enc.Write([]byte(msg.PlainText))
-			_ = enc.Close()
+			_, err = enc.Write([]byte(msg.PlainText))
+			if err != nil {
+				return nil, Error.Wrap(err)
+			}
+
+			defer func() { err = errs.Combine(err, enc.Close()) }()
```
Comment (ifraixedes) [inline internal/post/message.go]: This defer should be just after the `enc` initialization for guaranteeing that always gets closed.

File: internal/post/message.go [hunk 5]
```diff
@@ -93,18 +105,26 @@ func (msg *Message) Bytes() []byte {
...
 		fmt.Fprintf(&body, "Content-Transfer-Encoding: quoted-printable\r\n\r\n")
 
 		enc := quotedprintable.NewWriter(&body)
-		_, _ = enc.Write([]byte(msg.PlainText))
-		_ = enc.Close()
+		if _, err := enc.Write([]byte(msg.PlainText)); err != nil {
+			return nil, Error.Wrap(err)
+		}
+
+		if err := enc.Close(); err != nil {
```
Comment (ifraixedes) [inline internal/post/message.go]: Same reason than before, defer the close after the `enc` initialization.

Comment (simongui) [reply internal/post/message.go]: @ifraixedes Can I get a final review please? :) Thanks!

Comment (ifraixedes) [reply internal/post/message.go]: Only this one is missing to be deferred after `enc := ...` in the line 102 using the `errs.Combine` as you used above.

Comment (simongui) [reply internal/post/message.go]: Oops missed that one. Good catch. Done!

File: internal/post/sender.go [hunk 6]
```diff
@@ -76,7 +76,12 @@ func (sender *SMTPSender) SendEmail(msg *Message) error {
 		err = errs.Combine(err, data.Close())
 	}()
 
-	_, err = data.Write(msg.Bytes())
+	mess, err := msg.Bytes()
+	if err != nil {
+		return err
```
Comment (ifraixedes) [inline internal/post/sender.go]: This should be `return Error.Wrap(err)` however all the errors of this function are returned as they so I would suggest to leave it as it's now and open an issue in Jira to revisit all the methods and functions of this package to check the ones that should be to be `Error.Wrap(..)` which they will probably be the most of them

**Pull request merged**